cmd/asset-syncer/utils.go

```diff
@@ -871,3 +882,21 @@ func (f *fileImporter) fetchAndImportFiles(name string, repo Repo, cv models.Cha
 	// entry if digest has changed
 	return f.manager.insertFiles(chartID, chartFiles)
 }
+
+// Check if two URL strings are in the same domain.
+// Return true if so, and false otherwise or when an error occurs
+func compareURLDomains(url1Str, url2Str string) bool {
+	url1, err := url.ParseRequestURI(url1Str)
+	if err != nil {
+		return false
+	}
+	url2, err := url.ParseRequestURI(url2Str)
+	if err != nil {
+		return false
+	}
+
+	a := url1.Hostname()
+	b := url2.Hostname()
+	// return url1.Hostname() == url2.Hostname()
```

Comment:
Guessing this is what you intended to return rather than using the `a` and `b` unnecessarily?

But related to this, I reckon we should also ensure that the proto is the same (ie. both https). If the index.yaml was listed as https://example.com/index.yaml but referred to an icon at http://example.com/icons/foo.png, I'd hope we wouldn't assume it's the same host (ie. same host name but different proto). Thoughts?

Reply:
Yep, just a leftover when running the debugger :P 
Good catch! We have to verify the URL scheme as well, of course! 

By the way, there is an interesting go testing feature that will get released in 1.17: Fuzzing tests: https://blog.golang.org/fuzz-beta
Testing this method would be an interesting use case indeed.